alacritty/src/input.rs

```diff
@@ -613,7 +694,9 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
 
     /// Process a received character.
     pub fn received_char(&mut self, c: char) {
-        if *self.ctx.suppress_chars() {
+        if *self.ctx.suppress_chars()
+            || self.ctx.terminal().mode().contains(TermMode::KEYBOARD_MOTION)
```

Comment:
Related to https://github.com/alacritty/alacritty/pull/3315#discussion_r377914113. Is this the best place for this logic? There are other things we need to suppress as well, e.g. arrow keys. Generally, it'd be nice if our modes were a bit more encapsulated from each other. Why is `received_char` even being called in `TermMode::KEYBOARD_MOTION` for example?

Reply:
We don't need to suppress arrow keys. Arrow keys are bindings that send character events, they're the same as all other bindings. We do not want to block all bindings in keyboard mode.

This code is solely to block out `ReceivedCharacter` events. And this is absolutely the correct place for this and the only place in fact.